Assets/Scripts/Score/IScoreSystem.cs

```diff
@@ -0,0 +1,7 @@
+public interface ScoreSystem 
+{
+
+    void OnGameOver();
+    void OnGameOver(int Score);
+    void AddPoints(string value);
+}
```

Comment:
Naam is niet zelfde als file. maak het IScoreSystem

De OnGameOver kan weg. AddPoints is wel nessicary.

Wat je kan doen is de IScoreSystem generic maken. zodat je bijv `IScoreSystem<T>` hebt
Dan hebben `[SerializeField] private T _value;` en een `[SerializeField] private UnityEvent<T> _onValueChanged` dat we een event hebben die de nieuwe geupdate value return.